addons/sms/tests/test_sms_composer.py

```diff
@@ -0,0 +1,118 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.sms.tests.common import SMSCommon, SMSCase
+from odoo.tests import tagged
+from odoo.tools import html2plaintext, plaintext2html
+
+
+@tagged('at_install')
+class TestSMSComposerComment(SMSCommon, SMSCase):
+    """ Test behaviors that are overridden when other modules
+    are installed (e.g., mass_mailing). In these cases,
+    test_mail_sms or test_mail_full should be used."""
+
+    def test_message_post_sms_vs_notification(self):
+        """Check that the conversion of html to plain text does remove links
+
+        This is necessary when an SMS is sent from message_post with sms type
+        and not from _message_sms. In this case, it can be expected to receive html
+        that should be interpreted as such instead of escaped before being sent.
+
+        Note that as it is not simple nor desirable to inline replace a link such as
+        `<a href="href">Here</a>` to `href` in the sms, we keep the footnote behavior
+        of html2plaintext in this case (see second case tested).
+        """
+        cases = [
+            (
+                'Hello there, check this awesome <b>app</b> I found:<br/>https://odoo.com',  # not a `a` link in source
+                '<p>Hello there, check this awesome <b>app</b> I found:<br>https://odoo.com</p>',
+                'Hello there, check this awesome *app* I found:\nhttps://odoo.com'
+            ), (
+                'Hello there, check this awesome <b>app</b> I found:<br/><a href="https://odoo.com">Here</a>',   # a link
+                '<p>Hello there, check this awesome <b>app</b> I found:<br><a href="https://odoo.com">Here</a></p>',
+                'Hello there, check this awesome *app* I found:\nHere [1]\n\n\n[1] https://odoo.com'  # keep all information
+            )
+        ]
+
+        for message_content, expected_notification_content, expected_sms_content in cases:
+            with self.subTest(message_content=message_content):
+                with self.with_user('admin'), self.mockSMSGateway():
+                    message = self.env.user.partner_id.message_post(
```

Comment:
Can't we also test '_message_sms' so that the three "entry points" are tested ? _sms, post and composer ?